docs/spec_draft.md

```diff
@@ -1029,6 +1030,52 @@ specification. Numeric precision is implementation-defined.
 
 [Back to Ops](#index-of-documented-ops)
 
+## stablehlo.transpose
+
+`stablehlo.transpose(operand, permutation) -> result`
+
+### Semantics
+
+Permutes the dimensions of `operand` tensor using a permutation of its shape
+`permutation` and produces a `result` tensor. The $i^{th}$ dimension of the
+`result` is given by `permutation`$[i]^{th}$ dimension of the `operand`.
```

Comment:
This section doesn't really explain how to obtain the `result` tensor from the `operand` tensor. "The X dimension of a tensor is given by the Y dimension of another tensor" sounds fancy but is not actionable.

Here, I have the same feedback as in #170. Let's define op semantics from first principles, using only the definitions in this spec and college maths. Before proceeding with that, let's first expand our vocabulary - something which I took a stab at in #193.